java/java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java

```diff
@@ -227,13 +227,44 @@ public static ErrorDescription unboxingConditional(HintContext ctx) {
     }
     
     
-    @TriggerPattern("switch ($select) { case $cases$; }")
+    @TriggerTreeKind({Kind.SWITCH, Kind.SWITCH_EXPRESSION})
     public static ErrorDescription switchExpression(HintContext ctx) {
-        TreePath select = ctx.getVariables().get("$select");
+        ExpressionTree selector;
+        List<? extends CaseTree> cases;
+        Tree swtch = ctx.getPath().getLeaf();
+
+        switch (swtch.getKind()) {
+            case SWITCH -> {
+                SwitchTree st = (SwitchTree) swtch;
+                selector = st.getExpression();
+                cases = st.getCases();
+            }
+            case SWITCH_EXPRESSION -> {
+                SwitchExpressionTree st = (SwitchExpressionTree) swtch;
+                selector = st.getExpression();
+                cases = st.getCases();
+            }
+            default -> throw new IllegalStateException("Unexpected tree kind: " + swtch.getKind());
+        }
+
+        TreePath select = new TreePath(ctx.getPath(), selector);
         TypeMirror m = ctx.getInfo().getTrees().getTypeMirror(select);
         if (m == null || m.getKind() != TypeKind.DECLARED) {
             return null;
         }
+
+        boolean hasNullCase = cases.stream()
+                                   .flatMap(ct -> ct.getLabels().stream())
+                                   .filter(clt -> clt.getKind() == Kind.CONSTANT_CASE_LABEL)
+                                   .map(ctl -> ((ConstantCaseLabelTree) ctl).getConstantExpression())
+                                   .filter(expr -> expr.getKind() == Kind.NULL_LITERAL)
+                                   .findAny()
+                                   .isPresent();
```

Comment:
nitpick: one of my jackpot rules matched here, shorter version which combines the last three stages:

```java
        boolean hasNullCase = cases.stream()
                                   .flatMap(ct -> ct.getLabels().stream())
                                   .filter(clt -> clt.getKind() == Kind.CONSTANT_CASE_LABEL)
                                   .map(ctl -> ((ConstantCaseLabelTree) ctl).getConstantExpression())
                                   .anyMatch(expr -> expr.getKind() == Kind.NULL_LITERAL);
```